src/access/AccessRestrictionProvider.cpp

```diff
@@ -0,0 +1,252 @@
+    for (auto & entry : entries)
+    {
+        if (!IsEntryValid(entry))
+        {
+            ChipLogError(DataManagement, "AccessRestrictionProvider: invalid entry");
+            return CHIP_ERROR_INVALID_ARGUMENT;
+        }
+
+        Entry updatedEntry       = entry;
+        updatedEntry.fabricIndex = fabricIndex;
+        updatedEntries.push_back(updatedEntry);
+    }
+
+    mFabricEntries[fabricIndex] = std::move(updatedEntries);
+
+    for (Listener * listener = mListeners; listener != nullptr; listener = listener->mNext)
+    {
+        listener->RestrictionListChanged(fabricIndex);
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+bool AccessRestrictionProvider::StandardAccessRestrictionExceptionChecker::AreRestrictionsDisallowed(const SubjectDescriptor & subjectDescriptor,
+                                                                         const RequestPath & requestPath)
+{
+    if (requestPath.endpoint == 0 ||
+        requestPath.cluster == app::Clusters::NetworkCommissioning::Id ||
+        requestPath.cluster == app::Clusters::Descriptor::Id)
+    {
+        return true;
+    }
+
+    return false;
+}
+
+CHIP_ERROR AccessRestrictionProvider::CheckForCommissioning(const SubjectDescriptor & subjectDescriptor,
+                                                            const RequestPath & requestPath)
+{
+    if (mExceptionChecker.AreRestrictionsDisallowed(subjectDescriptor, requestPath))
+    {
+        return CHIP_NO_ERROR;
+    }
+
+    return DoCheck(mCommissioningEntries, subjectDescriptor, requestPath);
+}
+
+CHIP_ERROR AccessRestrictionProvider::Check(const SubjectDescriptor & subjectDescriptor, const RequestPath & requestPath)
+{
+    return DoCheck(mFabricEntries[subjectDescriptor.fabricIndex], subjectDescriptor, requestPath);
+}
+
+CHIP_ERROR AccessRestrictionProvider::DoCheck(const std::vector<Entry> entries, const SubjectDescriptor & subjectDescriptor,
+                                              const RequestPath & requestPath)
+{
+    ChipLogProgress(DataManagement, "AccessRestrictionProvider: action %d", static_cast<int>(requestPath.requestType));
+
+    if (requestPath.requestType == RequestType::kRequestTypeUnknown)
+    {
+        ChipLogError(DataManagement, "AccessRestrictionProvider: RequestPath type is unknown");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    // wildcard event subscriptions are allowed since wildcard is only used when setting up the subscription and
+    // we want that request to succeed (when generating the report, this method will be called with the specific
+    // event id). All other requests require an entity id
+    if (!requestPath.entityId.has_value())
+    {
+        if (requestPath.requestType == RequestType::kEventReadOrSubscribeRequest)
+        {
+            return CHIP_NO_ERROR;
+        }
+        else
+        {
+            return CHIP_ERROR_INVALID_ARGUMENT;
+        }
+    }
+
+    for (auto & entry : mFabricEntries[subjectDescriptor.fabricIndex])
```

Comment:
```suggestion
    for (auto & entry : entries)
```

The fact that this did not fail unit tests means that the commissioning ARL is NOT tested. Please fix

Reply:
fixed.  I will spend time on better tests next.